pkg/cmd/repo/autolink/list_test.go

```diff
@@ -0,0 +1,256 @@
+package autolink
+
+import (
+	"bytes"
+	"net/http"
+	"testing"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/internal/browser"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/httpmock"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/google/shlex"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewCmdList(t *testing.T) {
+	tests := []struct {
+		name         string
+		input        string
+		output       listOptions
+		wantErr      bool
+		wantExporter bool
+		errMsg       string
+	}{
+		{
+			name:   "no argument",
+			input:  "",
+			output: listOptions{},
+		},
+		{
+			name:   "web flag",
+			input:  "--web",
+			output: listOptions{WebMode: true},
+		},
+		{
+			name:         "json flag",
+			input:        "--json id",
+			output:       listOptions{},
+			wantExporter: true,
+		},
+		{
+			name:    "invalid json flag",
+			input:   "--json invalid",
+			wantErr: true,
+			errMsg: heredoc.Doc(`
+				Unknown JSON field: "invalid"
+				Available fields:
+				  id
+				  isAlphanumeric
+				  keyPrefix
+				  urlTemplate`),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ios, _, _, _ := iostreams.Test()
+			f := &cmdutil.Factory{
+				IOStreams: ios,
+			}
+
+			argv, err := shlex.Split(tt.input)
+			require.NoError(t, err)
+
+			var gotOpts *listOptions
+			cmd := newCmdList(f, func(opts *listOptions) error {
+				gotOpts = opts
+				return nil
+			})
+
+			cmd.SetArgs(argv)
+			cmd.SetIn(&bytes.Buffer{})
+			cmd.SetOut(&bytes.Buffer{})
+			cmd.SetErr(&bytes.Buffer{})
+
+			_, err = cmd.ExecuteC()
+			if tt.wantErr {
+				require.EqualError(t, err, tt.errMsg)
+				return
+			}
+
+			require.NoError(t, err)
+			assert.Equal(t, tt.output.WebMode, gotOpts.WebMode)
+			assert.Equal(t, tt.wantExporter, gotOpts.Exporter != nil)
+		})
+	}
```

Comment:
The `return` here feels weird to me... I think it's a bit easier to read/understand if written with a conditional instead of an early return:

```suggestion
			if tt.wantErr {
				require.EqualError(t, err, tt.errMsg)
			} else {
			        require.NoError(t, err)
			        assert.Equal(t, tt.output.WebMode, gotOpts.WebMode)
			        assert.Equal(t, tt.wantExporter, gotOpts.Exporter != nil)
		        }
		})
	}
```